### ISSUE TITLE
Add a way to configure vim mode

### ISSUE THREAD

**Issue by kerryland** [Zettlr/Zettlr#4643]
### Description

People who want to use vim-mode for editing expect to be able configure vim's default behaviour.

When editing markdown files it is very helpful to be able to reconfigure up and down keys so that they move through "screen" lines, not "physical" lines. ie: I need to be able to configure this (at least):
```
nnoremap <Down> gj
nnoremap <Up> gk
```

### Proposed Changes

Provide a way to execute vim commands when vim-mode begins. In "real" `vim` this is handled by a `.vimrc`. Obsidian, who also use `codewarrior-vim` have a `.obsidian.vimrc` file via [this plugin](https://github.com/esm7/obsidian-vimrc-support)

Another vimrc plugin for codewarrior can be found [here](https://github.com/personalizedrefrigerator/joplin-plugin-codemirror-vimrc)

It would be awesome if this code could be adapted for Zettlr.

### Caveats

This would have no impact on any other functionality.

### Do you Wish to Attempt Implementing this Yourself?

No

### Zettlr Version

Stable (most recent version)

### Your Platform

- [X] Windows
- [ ] macOS
- [ ] Linux

### Operating System Version

win10

**Comment by kyaso** [comment]
Realistically speaking, adding full vimrc support is most probably not a priority (let alone possibility) as the present moment.

However, regarding this:

> When editing markdown files it is very helpful to be able to reconfigure up and down keys so that they move through "screen" lines, not "physical" lines.

In my opinion, we could even go as far as to simply hardcode those two bindings. The [Vim mode extension](https://github.com/replit/codemirror-vim) which Zettlr uses, supports easy map/unmap of keys. For example:

```
Vim.map('j', 'gj')
Vim.map('k', 'gk')
```
I don't see a reason why anyone would _not_ prefer the `gj/gk` bindings over the default ones, so this would a good compromise.

**Comment by kerryland** [comment]
`15j` would translate to 15 x `gj`, which is what I would want to happen when editing text. I'm happy with this compromise, but I will deny this in a court of law if hard-core vim fanatics ever link me to this decision.

**Comment by bloomingseed** [comment]
> Realistically speaking, adding full vimrc support is most probably not a priority (let alone possibility) at the present moment.
> 
> However, regarding this:
> 
> > When editing markdown files it is very helpful to be able to reconfigure up and down keys so that they move through "screen" lines, not "physical" lines.
> 
> In my opinion, we could even go as far as to simply hardcode those two bindings. The [Vim mode extension](https://github.com/replit/codemirror-vim) which Zettlr uses, supports easy map/unmap of keys. For example:
> 
> ```
> Vim.map('j', 'gj')
> Vim.map('k', 'gk')
> ```
> 
> I don't see a reason why anyone would _not_ prefer the `gj/gk` bindings over the default ones (at least in the context of Zettlr), so this would be a good compromise.
> 
> **Edit**: Well, the new bindings might mess with certain move commands such as `15j`. But I wonder if there is anyone out there who relies on those particular movement commands; plus Zettlr doesn't display line numbers, so those commands are not that useful anyways (at least in my eyes...)

__tldr__: Open a document in normal mode --> `: map j gj`, `:map k gk`

Well, the new settings don't persist so I still have to re-enter the commands every time I open Zettlr.

**Comment by kyaso** [comment]
> **tldr**: Open a document in normal mode --> `:map j gj`, `:map k gk`
> 
> Or, just update Zettlr to version 3.3.0 or above ([#5479](https://github.com/Zettlr/Zettlr/pull/5479))

Yes, this is now the default behavior since v3.3.0!
